contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveScan.java

```diff
@@ -319,16 +319,12 @@ public ScanStats getScanStats() {
         estRowCount = data/1024;
       }
 
-      // Hive's native reader is neither memory efficient nor fast. If the rowcount is below
-      // HIVE_SERDE_SCAN_OVERHEAD_FACTOR, make sure it is at least HIVE_SERDE_SCAN_OVERHEAD_FACTOR to enable the planner
-      // to choose HiveDrillNativeParquetScan. Due to the project on top of HiveDrillNativeParquetScan, we end up
-      // choosing the HiveScan instead of HiveDrillNativeParquetScan if the cost is too low.
-      if (estRowCount <= HIVE_SERDE_SCAN_OVERHEAD_FACTOR) {
-        estRowCount = HIVE_SERDE_SCAN_OVERHEAD_FACTOR;
-      }
+      // Hive's native reader is neither memory efficient nor fast. Increase the CPU cost
+      // by a factor to let the planner choose HiveDrillNativeScan over HiveScan with SerDes.
+      float cpuCost = 1 * HIVE_SERDE_SCAN_OVERHEAD_FACTOR;
```

Comment:
I am thinking this cost should include the number of columns as well...suppose there are lots of columns(e.g 500), we want both types of plans to be equally penalized.